openssl/src/bn.rs

```diff
@@ -745,6 +745,25 @@ impl BigNumRef {
         }
     }
 
+    /// Places the result of `self^p mod m` in `self`. This makes it possible to avoid allocating
+    /// a new BigNum for each exponentiation. Useful for computations like `(((g^x_1 mod n)^x_2 mod n)^x_3 mod n)...`
+    pub fn mod_exp_mut(
```

Comment:
I'm not sure about the naming here - `mod_exp_into_self` is another option but a bit more verbose?

Reply:
Alright, `mod_exp_into_self` is better.